dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx

```diff
@@ -60,102 +56,119 @@ export const Gauge = ({ visualizations, layout, config }: any) => {
   const orientation = dataConfig?.chartStyles?.orientation || OrientationDefault;
   const legendPlacement = dataConfig?.chartStyles?.legendPlacement || LegendPlacement;
 
-  const isEmptyPlot = !seriesLength;
+  const isEmptyPlot = !seriesLength || isEmpty(queriedVizData);
+
+  const timestampField = find(fields, (field) => field.type === 'timestamp');
+
+  if (dataConfig.span && dataConfig.span.time_field && timestampField) {
+    xaxes = [timestampField, ...xaxes];
+  }
+
+  const xaxesLength = xaxes.length;
+
+  const getAggValue = (metric: any) => {
+    return metric.alias ? metric.alias : metric.aggregation + '' + '(' + metric.name + ')';
```

Comment:
use template literals instead

Reply:
For this used the common function created by Dipra in his PR